### PR TITLE
[enhancement] address new warning in LocalOutlierFactor in sklearn1.6

### PR DESCRIPTION
**Pull request by icfaust**
Adds conformance for the new test: 
test_lof.py::test_lof_duplicate_samples

### CODE REVIEW

File: sklearnex/neighbors/_lof.py [hunk 1]
```diff
@@ -97,6 +97,13 @@ def _onedal_fit(self, X, y, queue=None):
                 self.negative_outlier_factor_, 100.0 * self.contamination
             )
 
+        if sklearn_check_version("1.6"):
```
Comment (samir-nasibli) [inline sklearnex/neighbors/_lof.py:100]: Could you please reference where and why is it needed?

**Review by samir-nasibli — COMMENTED** [review]
General question. 1.6 is not stable release version. Should we address this cases, even if we don't have this on our matrices?

File: sklearnex/neighbors/_lof.py (hunk 1 above)
Comment (icfaust) [reply sklearnex/neighbors/_lof.py:100]: scikit-learn#28773 which is already in planned inclusion in sklearn 1.6: https://scikit-learn.org/dev//whats_new/v1.6.html

Comment (Alexsandruss) [reply sklearnex/neighbors/_lof.py:100]: I guess Samir meant this sklearn issue/PR should be referenced in comment for better documented code. Anyway, I think it is good to start doing it in sklearnex for future explainability of code.

**Review by Alexsandruss — APPROVED** [review]

**Pull request merged**